test/fixtures/cookbooks/test/recipes/filter_lines.rb

```diff
@@ -0,0 +1,125 @@
+directory '/tmp'
+
+template '/tmp/dangerfile' do
+end
+
+filter_lines 'Do nothing' do
+  path '/tmp/dangerfile'
+  filter proc { |current| current }
+end
+
+template '/tmp/reverse' do
+  source 'dangerfile.erb'
+end
+
+filter_lines 'Reverse line text' do
```

Comment:
Arbitrary in line proc.  Can do anything with the array as long as a new array is returned.